src/backend/profile/profile.service.ts

```diff
@@ -100,3 +101,112 @@ export async function createUserProfile(profileData: CreateProfilePayload) {
   })
   return createdUser
 }
+
+async function updateUserCity(userId: string, newCity: City): Promise<void> {
+  if (!newCity) return
+
+  await prisma.profile.update({
```

Comment:
**nitpick:** Instead of `connectOrCreate`, use `upsert`, so it will update record if found, or create if does not exist